pkg/forwarder/forwarder.go

```diff
@@ -0,0 +1,57 @@
+package forwarder
+
+import (
+	"context"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+type Forwarder struct {
+	router    *message.Router
+	publisher message.Publisher
+}
+
+func NewForwarder(publisher message.Publisher, subscriber message.Subscriber, logger watermill.LoggerAdapter, config Config) (*Forwarder, error) {
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &Forwarder{router, publisher}
+
+	router.AddNoPublisherHandler(
```

Comment:
in theory it should work with `AddNoPublisherHandler` - in this case we don't need to inject publisher, we can just return message for publishing

Reply:
ah ok, it's not possible because topic is in the envelope, nvm